From 8b6a6bd050ff6a1e3c04bcb11e357433f600d572 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Fri, 28 Oct 2022 10:35:42 +0300
Subject: [PATCH 23/49] fdts: s32: add s32cc-crypto.dtsi file

The newly introduced DT contains information about s32's Messaging Units
(MUs). The four instances facilitate the communication between the A53
cores and HSE. Include s32cc-crypto.dtsi in the dts if HSE_SECBOOT is
enabled.

Issue: ALB-8989
Upstream-Status: Pending 

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/s32cc-crypto.dtsi     | 79 ++++++++++++++++++++++++++++++++++++++
 fdts/s32cc.dtsi            |  3 ++
 plat/nxp/s32/s32_common.mk |  1 +
 3 files changed, 83 insertions(+)
 create mode 100644 fdts/s32cc-crypto.dtsi

diff --git a/fdts/s32cc-crypto.dtsi b/fdts/s32cc-crypto.dtsi
new file mode 100644
index 000000000..f53057d0a
--- /dev/null
+++ b/fdts/s32cc-crypto.dtsi
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	soc {
+		hse: crypto {
+			compatible = "simple-bus";
+			#address-cells = <2>;
+			#size-cells = <2>;
+			#interrupt-cells = <3>;
+			memory-region = <&hse_reserved>;
+			ranges;
+
+			mu0b@40210000 {
+				compatible = "nxp,s32cc-hse";
+				reg = <0x0 0x40210000 0x0 0x1000>,
+				      <0x0 0x22c00000 0x0 0x1000>;
+				reg-names = "hse-regs",
+					    "hse-desc";
+				interrupts = <GIC_SPI 103 IRQ_TYPE_EDGE_RISING>, /* GIC 135 */
+					     <GIC_SPI 104 IRQ_TYPE_EDGE_RISING>, /* GIC 136 */
+					     <GIC_SPI 105 IRQ_TYPE_EDGE_RISING>; /* GIC 137 */
+				interrupt-names = "hse-ack",
+						  "hse-rx",
+						  "hse-err";
+				status = "disabled";
+			};
+
+			mu1b@40211000 {
+				compatible = "nxp,s32cc-hse";
+				reg = <0x0 0x40211000 0x0 0x1000>,
+				      <0x0 0x22c01000 0x0 0x1000>;
+				reg-names = "hse-regs",
+					    "hse-desc";
+				interrupts = <GIC_SPI 106 IRQ_TYPE_EDGE_RISING>, /* GIC 138 */
+					     <GIC_SPI 107 IRQ_TYPE_EDGE_RISING>, /* GIC 139 */
+					     <GIC_SPI 108 IRQ_TYPE_EDGE_RISING>; /* GIC 140 */
+				interrupt-names = "hse-ack",
+						  "hse-rx",
+						  "hse-err";
+				status = "disabled";
+			};
+
+			mu2b@40212000 {
+				compatible = "nxp,s32cc-hse";
+				reg = <0x0 0x40212000 0x0 0x1000>,
+				      <0x0 0x22c02000 0x0 0x1000>;
+				reg-names = "hse-regs",
+					    "hse-desc";
+				interrupts = <GIC_SPI 109 IRQ_TYPE_EDGE_RISING>, /* GIC 141 */
+					     <GIC_SPI 110 IRQ_TYPE_EDGE_RISING>, /* GIC 142 */
+					     <GIC_SPI 111 IRQ_TYPE_EDGE_RISING>; /* GIC 143 */
+				interrupt-names = "hse-ack",
+						  "hse-rx",
+						  "hse-err";
+				status = "okay";
+			};
+
+			mu3b@40213000 {
+				compatible = "nxp,s32cc-hse";
+				reg = <0x0 0x40213000 0x0 0x1000>,
+				      <0x0 0x22c03000 0x0 0x1000>;
+				reg-names = "hse-regs",
+					    "hse-desc";
+				interrupts = <GIC_SPI 112 IRQ_TYPE_EDGE_RISING>, /* GIC 144 */
+					     <GIC_SPI 113 IRQ_TYPE_EDGE_RISING>, /* GIC 145 */
+					     <GIC_SPI 114 IRQ_TYPE_EDGE_RISING>; /* GIC 146 */
+				interrupt-names = "hse-ack",
+						  "hse-rx",
+						  "hse-err";
+				status = "disabled";
+			};
+		};
+	};
+};
diff --git a/fdts/s32cc.dtsi b/fdts/s32cc.dtsi
index 5d3feef71..b052f7375 100644
--- a/fdts/s32cc.dtsi
+++ b/fdts/s32cc.dtsi
@@ -12,6 +12,9 @@
 #include <dt-bindings/phy/phy.h>
 #include <dt-bindings/reset/s32gen1-scmi-reset.h>
 #include <dt-bindings/reset/s32gen1-wkpu.h>
+#if defined(HSE_SECBOOT)
+#include "s32cc-crypto.dtsi"
+#endif
 
 / {
 	interrupt-parent = <&gic>;
diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index d5b2d57d5..723621562 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -188,6 +188,7 @@ $(eval $(call add_define_val,S32_SET_NEAREST_FREQ,$(S32_SET_NEAREST_FREQ)))
 # Process HSE_SECBOOT flag
 ifneq (${HSE_SECBOOT},)
 $(eval $(call add_define,HSE_SECBOOT))
+$(eval $(call add_define,HSE_MU_INST,4))
 endif
 
 # Reserve some space at the end of SRAM for external apps and include it
-- 
2.25.1

